src/Microsoft.ML.Sweeper/Algorithms/Grid.cs

```diff
@@ -173,7 +177,12 @@ public override ParameterSet[] ProposeSweeps(int maxSweeps, IEnumerable<IRunResu
                     if (!AlreadyGenerated(_cache[iPerm], prevParamSets))
                         break;
                 }
-                result.Add(_cache[iPerm]);
+
+                if(!_cache.Contains(_cache[iPerm]))
```

Comment:
>_cache [](start = 20, length = 6)

did you mean to check if result Contains it?

Reply:
Oh, whoops! I think you are correct. I'll update that.

Reply:
Also, this would avoid returning duplicates, which is good, but we'd still have duplicates stored in the _cache.  Might be just good practice to avoid storing the result directly in the _cache, without checking for duplicates. 

---
In reply to: [195774135](https://github.com/dotnet/machinelearning/pull/358#discussion_r195774135) [](ancestors = 195774135)

Reply:
This is probably way off, but would something like this work when initializing the `_cache`?

```
_cache = (new ParameterSet[_nGridPoints]).Distinct().ToArray();
```

Reply:
probably not, because the values are generated later, by calling CreateParamSet. 

thinking more about it _cache is an array, we'd have to iterate through it ourselves. Let's skip it, and just filter the duplicates out of the result returned. 

---
In reply to: [195781506](https://github.com/dotnet/machinelearning/pull/358#discussion_r195781506) [](ancestors = 195781506)